dwave/gate/simulator/simulator.pyx

```diff
@@ -178,6 +181,22 @@ def _measure(op, state, targets):
         op._measured_qubit_indices.append(t)
 
 
+def _fast_sample(
+    qubit: int,
+    state: np.typing.NDArray,
+    collapse_state: bool = True,
+    little_endian: bool = False,
+) -> int:
+    cdef int num_qubits = round(np.sqrt(state.shape[0]))
+
+    return measurement_computational_basis(
+        num_qubits,
+        state,
+        qubit,
+        little_endian=little_endian
+    )
```

Comment:
Should `collapse_state` be passed to the `apply_operator` arg? Feel free to rename it as well to the latter, if you think that makes more sense, to keep it to a single name.
```suggestion
    return measurement_computational_basis(
        num_qubits,
        state,
        qubit,
        little_endian=little_endian,
        apply_operator=collapse_state,
    )
```

Reply:
Yes it should be. I think `collapse_state` is a good name for measurement, I just wanted to make it a bit more general sounding for the other stuff that the operator set functions allow.

Eventually this will be set to False when using `Measurement.sample()`, right?

Reply:
> Eventually this will be set to False when using Measurement.sample(), right?

Yes, it should be set to `False`, although currently this sampling is implemented separately ([here](https://github.com/dwavesystems/dwave-gate/blob/master/dwave/gate/tools/samples.py#L22-L47)), so there's no collapsing taking place anyway.